eng/psakefile.ps1

```diff
@@ -427,3 +427,27 @@ function Create-DocsEnv() {
         deactivate
     }
 }
+
+task check-licenses {
+    # Uses cargo-deny to verify that the linked components
+    # only use approved licenses
+    # https://github.com/EmbarkStudios/cargo-deny
+    Invoke-LoggedCommand -wd $repo.root {
+        cargo deny check licenses
+    }
+}
+
+task update-noticefile {
+    # use cargo-about to generate a notice file
+    # llvm special license is already in the template
+    # as it is a hidden transitive dependency.
+    # https://github.com/EmbarkStudios/cargo-about
+    $config = Join-Path $repo.root notice.toml
+    $template = Join-Path $repo.root notice.hbs
+    $notice = Join-Path $repo.root NOTICE.txt
+    Invoke-LoggedCommand -wd $repo.root {
+        cargo about generate --config $config --workspace --all-features --output-file $notice $template
```

Comment:
Does the `cargo-about` tool need to get installed separately? I don't see a place where that happens, but maybe I'm missing it.

Reply:
This is a manual process so you'd have to install it yourself. I don't know of a way to ask cargo if a subcommand exists.

Reply:
I think `cargo install --list` will show all of them, and then you could parse to check, but not really needed here since `cargo about` will fail with a reasonable message.